shared/js/toc.jsx

```diff
@@ -0,0 +1,559 @@
+      event.preventDefault();
+      var input = this.refs.siteUrl.getDOMNode();
+      input.value && this.props.handleAddUrlClick(input.value);
+    },
+
+    render: function() {
+      return (
+        <div className="room-panel-add-url">
+          <h2>{'Add a site to the room'}</h2>
+          <input placeholder="http://..." ref="siteUrl" type="text" />
+          <button onClick={this.handleClick}>{'Add site'}</button>
+        </div>
+      );
+    }
+  });
+
+  var RoomContentView = React.createClass({
+    propTypes: {
+      tiles: React.PropTypes.array
+    },
+
+    getDefaultProps: function() {
+      return {
+        tiles: []
+      };
+    },
+
+    render: function() {
+      return (
+        <div className="room-toc">
+          {
+            this.props.tiles.map(function(tile, index) {
+              return (
+                <TileView
+                  key={index}
+                  tile={tile} />
+              );
+            }, this)
+          }
+        </div>
+      );
+    }
+  });
+
+  var TileView = React.createClass({
+    statics: {
+      GOOGLE_SCREENSHOT: "https://www.googleapis.com/pagespeedonline/v1/runPagespeed?screenshot=true&strategy=desktop&url="
+    },
+
+    propTypes: {
+      tile: React.PropTypes.object.isRequired
+    },
+
+    getInitialState: function() {
+      return {
+        screenshot: null
+      };
+    },
+
+    componentWillMount: function() {
+      var xhr = new XMLHttpRequest();
+      var url = this.constructor.GOOGLE_SCREENSHOT + this.props.tile.location;
+
+      xhr.onreadystatechange = function() {
+        if (xhr.readyState === 4 && xhr.status === 200) {
+          var response = JSON.parse(xhr.response).screenshot;
+          console.info(JSON.parse(xhr.response).screenshot);
+          var src = "data:" + response.mime_type + ";base64,";
+          var base64 = response.data;
+          base64 = base64.replace(/\_/g, "/");
+          base64 = base64.replace(/\-/g, "+");
+          src += base64;
+          this.setState({
+            screenshot: src
+          });
+        }
+      }.bind(this);
+      xhr.open("GET", url, true);
+      xhr.send();
+    },
+
+    shouldComponentUpdate: function(nextProps, nextState) {
+      return nextState.screenshot !== this.state.screenshot;
+    },
+
+    render: function() {
+      return (
+        <div className="toc-tile">
+          <div className="room-user" data-name="Pau Masiá">
+            <span>{'P'}</span>
+          </div>
+          <img className="tile-screenshot" src={this.state.screenshot} />
+          <div className="tile-info">
+            <a
+              className="tile-name"
+              href={this.props.tile.location}
+              title={this.props.tile.description}>
+                {this.props.tile.description}
+            </a>
+            <h3 className="tile-url">{this.props.tile.location}</h3>
+          </div>
+        </div>
+      );
+    }
+  });
+
+  var SidebarView = React.createClass({
```

Comment:
xref: https://github.com/mozilla/loop/pull/329/files#r60665491

I think it would be good to de-fork this now, or to simplify things, it could be the first thing we land after this PR.